com.unity.render-pipelines.high-definition/Runtime/PostProcessing/PostProcessSystem.RenderGraph.cs

```diff
@@ -65,113 +188,235 @@ class AlphaCopyPassData
+                                (DynamicExposureData data, RenderGraphContext ctx) =>
+                                {
+                                    DoHistogramBasedExposure(data.parameters, ctx.cmd, ctx.resources.GetTexture(data.source),
+                                                                                       ctx.resources.GetTexture(data.prevExposure),
+                                                                                       ctx.resources.GetTexture(data.nextExposure),
+                                                                                       ctx.resources.GetTexture(data.exposureDebugData));
+                                });
+                        }
+                        else
+                        {
+                            passData.tmpTarget1024 = builder.CreateTransientTexture(new TextureDesc(1024, 1024, true, false)
+                            { colorFormat = GraphicsFormat.R16G16_SFloat, enableRandomWrite = true, name = "Average Luminance Temp 1024" });
+                            passData.tmpTarget32 = builder.CreateTransientTexture(new TextureDesc(32, 32, true, false)
+                            { colorFormat = GraphicsFormat.R16G16_SFloat, enableRandomWrite = true, name = "Average Luminance Temp 32" });
+
+                            builder.SetRenderFunc(
+                                (DynamicExposureData data, RenderGraphContext ctx) =>
+                                {
+                                    DoDynamicExposure(data.parameters, ctx.cmd, ctx.resources.GetTexture(data.source),
+                                                                                ctx.resources.GetTexture(data.prevExposure),
+                                                                                ctx.resources.GetTexture(data.nextExposure),
+                                                                                ctx.resources.GetTexture(data.tmpTarget1024),
+                                                                                ctx.resources.GetTexture(data.tmpTarget32));
+                                });
+                        }
+                    }
+
+                    if (hdCamera.resetPostProcessingHistory)
+                    {
+                        using (var builder = renderGraph.AddRenderPass<ApplyExposureData>("Apply Exposure", out var passData, ProfilingSampler.Get(HDProfileId.ApplyExposure)))
+                        {
+                            passData.source = builder.ReadTexture(source);
+                            passData.parameters = PrepareApplyExposureParameters(hdCamera);
+                            RTHandle prevExp;
+                            GrabExposureHistoryTextures(hdCamera, out prevExp, out _);
+                            passData.prevExposure = renderGraph.ImportTexture(prevExp);
+
+                            TextureHandle dest = renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```

Comment:
This is repeated in several places. We could probably do a helper function to reduce code footprint